### PR TITLE
Triggers Abstracted to Server

### PR DESCRIPTION
**Pull request by jarredhawkins**
- changes in transaction server
  - handles triggers completely differently
  - new trigger objects and libraries
  - new function to receive from triggerServer
- database library modifications
- triggers are separate HTTP server
  - 4 endpoints, setTrigger, startTrigger, cancelTrigger, runningTriggers, waitingTriggers

### CODE REVIEW

File: parent/build.sh [hunk 1]
```diff
@@ -34,12 +36,27 @@ docker build \
 
 cd ../quoteserver
 docker build \
+--add-host quoteserve.seng:172.20.0.1 \
```
Comment (bjschnell) [inline parent/build.sh]: Probably don't need this line if you are passing it as build-arg below.

File: transaction-server/database/user_database.go [hunk 2]
```diff
@@ -19,22 +18,27 @@ type UserDatabase interface {
 	GetFunds(string) (decimal.Decimal, error)
 	RemoveFunds(string, decimal.Decimal) error
 
-	AddStock(user string, stock string, shares int) error
-	GetStock(user string, stock string) (int, error)
-	RemoveStock(user string, stock string, amount int) error
+	AddStock(user string, stock string, shares decimal.Decimal) error
```
Comment (bjschnell) [inline transaction-server/database/user_database.go]: Just curious why did these need to change?

Comment (jarredhawkins) [reply transaction-server/database/user_database.go]: In the log page, the amount examples were listed as decimals: http://www.ece.uvic.ca/~seng462/ProjectWebSite/Commands.html

I know for the purpose of this he mentioned we don't have to worry about partial stocks, but I figured might as well make them all decimals. That way we also don't have to worry about int vs decimal types too.

File: parent/build.sh (hunk 1 above)
Comment (jarredhawkins) [reply parent/build.sh]: You're right lol

**Pull request merged**